### PR TITLE
use MakePodSpec consistently

### LINKED ISSUE #121234: Cleanup: Validate pod terminationGracePeriodSeconds and fix tests
xref: https://github.com/kubernetes/kubernetes/pull/119026/files#r1265687485

Apparently we do not validate that the `terminationGracePeriodSeconds` field is non-nil.  It has a default value, so it should never actually be nil, but because we never validated it in the past, we have tests that do not set it.

We should remove code which check for nil, like the above comment:

```
	var gracePeriod int64
	if spec.TerminationGracePeriodSeconds != nil {
		// this could happen in tests
		gracePeriod = *spec.TerminationGracePeriodSeconds
	}
```

This will cause a bunch of tests to fail.  Those tests SHOULD be setting this field properly.  See "k8s.io/kubernetes/pkg/api/service/testing" which has helper functions for the same problem in Service.

### PR DESCRIPTION
**Pull request by fraenkel**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

cleaning up some tests after MakePod/MakePodSpec were introduced

#### Which issue(s) this PR fixes:

Fixes #121234 

#### Special notes for your reviewer:

These changes were based on comments of the previous PR that was merged.

#### Does this PR introduce a user-facing change?
NONE

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

**Comment by fraenkel** [comment]
@thockin here is the cleanup

File: pkg/apis/apps/validation/validation_test.go [hunk 1]
```diff
@@ -2136,7 +2136,7 @@ func TestValidateDaemonSet(t *testing.T) {
 			Spec: apps.DaemonSetSpec{
 				Selector: &metav1.LabelSelector{MatchLabels: validSelector},
 				Template: api.PodTemplateSpec{
-					Spec: podtest.MakePodSpec(api.RestartPolicyNever, podtest.SetLabels(validSelector)),
+					Spec: podtest.MakePodSpec(podtest.SetRestartPolicy(api.RestartPolicyNever), podtest.SetLabels(validSelector)),
```
Comment (thockin) [inline pkg/apis/apps/validation/validation_test.go:2139]: This is great - it really calls out that the policy is important in this specific case

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by fraenkel** [comment]
the original failure is known https://github.com/kubernetes/kubernetes/pull/125691

**Pull request merged**